src/adm/filter.rs

```diff
@@ -121,12 +121,13 @@ pub fn spawn_updater(
         let mut tags = crate::tags::Tags::default();
         loop {
             {
+                metrics.incr("filter.adm.update_check").ok();
```

Comment:
I'd recommend injecting metrics into every arm of this match expression to get the whole picture for each update attempt.

```rs
    {
        match mfilter.read().await.requires_update(&storage_client).await {
            Ok(true) => {
                let mut filter = mfilter.write().await;
                match filter.update(&storage_client).await {
                    Ok(_) => {
                        metrics.incr("adm.filter.update.ok").ok();
                    }
                    Err(e) => {
                        filter.report(&e, &mut tags);
                        metrics.incr("adm.filter.update.error").ok();
                    }
                }
                // filter.update(&storage_client).await.unwrap_or_else(|e| {
                // metrics.incr("adm.filter.update.error").ok();
                // filter.report(&e, &mut tags);
                // });
            }
            Ok(false) => {
                metrics.incr("adm.filter.update.check.skip").ok();
            }
            Err(e) => {
                mfilter.read().await.report(&e, &mut tags);
                metrics.incr("adm.filter.update.check.error").ok();
            }
        }
    }
```
What do you think?

Reply:
thorough sounds good